queries_go19_test.go

```diff
@@ -346,6 +346,107 @@ SELECT @param2 = 'World'
 	})
 }
 
+func TestOutputINOUTStringParam(t *testing.T) {
```

Comment:
>ingParam [](http://example.com/codeflow?start=23&length=8)

what about non-string types like varbinary? Is there a scenario where the app passes a byte slice or array and we have to allocate a bigger one on return?

Reply:
Well, the code branch that switches to PLP encoding in case of OUTPUT parameter resides under the following case condition:

```
case typeBigVarBin, typeBigVarChar, typeBigBinary, typeBigChar,
                typeNVarChar, typeNChar, typeXml, typeUdt:
```

So it must work for VARBINARY as well.

Should I:

* Add tests for all (or some of) these types
* Isolate changed code so that it affects NVARCHAR only (or maybe more specific types)
* Leave it as is

?

Reply:
i think one test for a non-string output will suffice. It's helpful as both example code and clear documentation.

Reply:
Added and checked if it fails in case of missing fix, please see updated code. I deliberately didn't include NULL cases since they seem to be not supported yet.